src/components/generic-modal.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+    <button type="button" class="btn align-self-center m-0 p-0"
+        @click="showModal"
+    >
+        <slot name="activation-button"></slot>
+    </button>
+    <client-only>
+        <Teleport to="#modal-container" :key="modalId">
+            <div
+                v-show="open"
+                :class="{ 'modal-box': true, 'd-block': open, 'd-none': !open }"
+                :id="modalId"
+            >
+                <div class="modal-content d-flex align-items-end flex-column">
+                    <div class="modal-header-box d-grid w-100">
+                        <div class="modal-heading sidenav-text my-auto mx-0 align-self-center">
+                            {{ title }}
+                        </div>
+                        <div class="modal-close-btn m-auto d-block" @click="closeModal">
+                            <PhX size="22" color="white"/>
+                        </div>
```

Comment:
Use bootstrap button so the right styles are added (eg. we now don't get the right cursor displayed on hover)

replace
```xml
                        <div class="modal-close-btn m-auto d-block" @click="closeModal">
                            <PhX size="22" color="white"/>
                        </div>
 ```
 with
```xml
                        <button class="btn btn-icon text-white fs-3" @click="closeModal">
                            <PhX />
                        </button>
```

Reply:
Worked on this.